versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayDependencyUpdatesMojo.java

```diff
@@ -489,6 +484,11 @@ protected void validateInput() throws MojoExecutionException {
         validateGAVList(pluginDependencyExcludes, 3, "pluginDependencyExcludes");
         validateGAVList(pluginManagementDependencyIncludes, 3, "pluginManagementDependencyIncludes");
         validateGAVList(pluginManagementDependencyExcludes, 3, "pluginManagementDependencyExcludes");
+        if (getLog() != null
+                && allowAnyUpdates
+                && !(allowMajorUpdates && allowMinorUpdates && allowIncrementalUpdates)) {
+            getLog().warn("Assuming allowAnyUpdates false because one or more other \"allow\" switches is false.");
```

Comment:
`allowAnyUpdates` has default value as `true` so when user provide only one `allow*` switch this warning will be always displayed.
As now we ignore at all `allowAnyUpdates` maybe it is not needed.

Reply:
> As now we ignore at all allowAnyUpdates maybe it is not needed.

The problem is that we do not ignore `allowAnyUpdates` -- we ignore all the other `allow` options if `allowAnyUpdates` is `true`.

See: https://github.com/mojohaus/versions/blob/0fe821e400fe917b5fc342fe5712071dd0490473/versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayDependencyUpdatesMojo.java#L514

That is basically the reason why I created this PR.

Reply:
In other words, if the user only sets `allowMajorUpdates=false`, that option will be ignored because the user also must use `allowAnyUpdates=false` together with that option for it to work... Which is flawed in my opinion.

Reply:
I see that after this change `allowAnyUpdates` will be ignored - so it will be not important what user set - right?

Reply:
Yes, with a warning message.

Reply:
ok, but settings or not `allowAnyUpdates` will mo have any  effect only warning message will be or not displayed
So I think that warning is such case is redundant
I don't see any more usage of `allowAnyUpdates` ...

Reply:
I don't see it either.

Ideally, I would remove it.

Reply:
Do it. `allowAnyUpdates` is only used in this goal. Will be more clear.

Reply:
agree - we can wait a few days.
I would to release next version at next week - about 25 - 26 of May